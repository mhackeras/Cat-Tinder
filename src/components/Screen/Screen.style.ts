import {StyleSheet} from 'react-native';
import {colors} from '../../res/colors';

export default StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: colors.background,
  },
  container: {
    flex: 1,
  },
});
