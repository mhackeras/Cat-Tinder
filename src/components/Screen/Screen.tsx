import React from 'react';
import {View} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import styles from './Screen.style';

interface Props {
  children: React.ReactNode;
}

const Screen = ({children}: Props) => {
  return (
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>{children}</View>
    </SafeAreaView>
  );
};

export default Screen;
