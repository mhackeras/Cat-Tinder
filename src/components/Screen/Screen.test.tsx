import React from 'react';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {View} from 'react-native';
import {render} from '@testing-library/react-native';

import Screen from './Screen';

describe('Screen', () => {
  it('renders a regular screen', () => {
    const tree = render(
      <SafeAreaProvider>
        <Screen>
          <View />
        </Screen>
      </SafeAreaProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
