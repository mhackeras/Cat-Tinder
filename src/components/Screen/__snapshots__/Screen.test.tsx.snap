// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen renders a regular screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
